bundle-workflow/src/test_workflow/test_cluster.py

```diff
@@ -5,20 +5,29 @@
 # compatible open source license.
 
 import abc
+from contextlib import contextmanager
+from test_workflow.perf_test_cluster import PerfTestCluster
 
 
 class TestCluster(abc.ABC):
     """
     Abstract base class for all types of test clusters.
     """
-
+    @staticmethod
     @abc.abstractmethod
-    def create(self):
+    @contextmanager
+    def create(cluster_type, *args):
```

Comment:
TestCluster should not know about its child, `PerfTestCluster`.  inherits TestCluster, then this code becomes:

```pythin
def create(cls, *args)
  cluster = cls(*args)
```

Reply:
When I am invoking `create` method like
`with PerfTestCluster.create(manifest, config, args.stack, security) as test_cluster_endpoint:`, it calls the `PerfTestCluster`'s `create` method.

Reply:
When I am invoking `create` method like
`with PerfTestCluster.create(manifest, config, args.stack, security) as test_cluster_endpoint:`. It calls the `PerfTestCluster`'s `create` method. I am not sure whether I am following the right path. Feel free to correct me here. The `create` method of `TestCluster` looks like below but won't get invoked, may be calling with `super` might help:
```python
        cluster = cls(*args)
        try:
            cluster.create()
            yield cluster.endpoint()
        finally:
            cluster.destroy()`

Reply:
When I am invoking `create` method like
`with PerfTestCluster.create(manifest, config, args.stack, security) as test_cluster_endpoint:`. It calls the `PerfTestCluster`'s `create` method. I am not sure whether I am following the right path. Feel free to correct me here. The `create` method of `TestCluster` looks like below but won't get invoked, may be calling with `super` might help:
```python
        cluster = cls(*args)
        try:
            cluster.create()
            yield cluster.endpoint()
        finally:
            cluster.destroy()`

Reply:
It will because it's a class method called `create`, not the instance method with the same name.

Reply:
I changed `PerfTestCluster`'s method to `create_cluster` since both methods of `TestCluster` and `PerfTestCluster` have the same name i.e. `create`. The class method gets overridden by the instance method, since it belongs to the `PerfTestCluster` class.